### ISSUE TITLE
Implement preset selection

### ISSUE THREAD

**Issue by mikessh** [antigenomics/vdjdb-server#16]
The following logic is now applicable to sequence-based search:
- It is not controlled by simply specifying substitution and indel count, etc
- Parameter setting is done in three steps
  1. A **preset** should be selected, e.g. `dummy` or `high-precision`. List of possible presets is here `SequenceSearcherPreset.ALLOWED_PRESETS`. Preset contains all the options. To load a preset call  e.g. `SequenceSearcherPreset preset = SequenceSearcherPreset.byName("dummy")`
  2. Users can manually change **search scope** (substitution and indel count) by calling e.g. `preset.withSearchParameters(2,1,1,2)`. This will speed-up/slow-down the search and increase/decrease the recall. Range is from `0,0,0,0` to `5,2,2,7` for the database browsing and (I'm unsure what performance we can reach on cdr3.net) to `4,1,1,4` for sample annotation.
  3. To change the **threshold** for the scoring system (threshold for `Alignment aln; aln.getScore()`) call `preset.withScoringThreshold(1000)`. Maybe implement with slide bar, range should be `-2e4` to `2e4`.

Basically we have 3 panels here: one for preset selection (one dropdown), one for search scope (four text fields) and one for threshold (one slider). All of them should be enabled for DB browsing. Only preset selection should be enabled for annotation and scope fixed to `4,1,1,4` in the version on cdr3.net (but not for standalone!!)

**Comment by mikessh** [comment]
Re-implemented everything in https://github.com/antigenomics/vdjdb-standalone/tree/feature/scoring-v2

Updated implementation:
- Preset names have changed a bit, see `ALLOWED_NAMES` in `SequenceSearcherPreset`
- You can either select by the preset name
- Or by precision (`SequenceSearcherPreset .byPrecision(float x)`) and recall(`SequenceSearcherPreset .byRecall(float x)`)

Interface suggestions:
- For browse panel sequence filter allow the following:
  1. Selecting preset from `ALLOWED_NAMES`
  2. Allow changing scope. Limit scope to `5,2,2,7` for `hamming` preset, `7,3,3,10` for other presets
  3. Allow selecting a pre-set by moving the precision/recall slide bar (range for both is [0,1]). Example for precision (update on slider move; precision and recall are connected):
  
  ``` groovy
  def precision = 0.8
  def metadata = new ScoringMetadataTable().getByPrecision(precision)
  // !!! set recall slider to metadata.getRecall() position
  SequenceSearcherPreset.byMetadata(metadata) // use this preset in search
  ```
- For "Annotate" (sample-based) part fix preset to `SequenceSearcherPreset.byName("high-precision")`

PS. Now exact matches are always reported, even if they are below the threshold.
